File: tests/fixtures/safe_pq_draft.js
// tests/fixtures/safe_pq_draft.js
//
// Positive non-match fixture for js/pq-vulnerable-crypto.
// generateKeyPair('ml-dsa'|'fn-dsa'|'hqc') must NOT fire: the rule matches
// on the literal algorithm argument, and draft NIST PQC names (FN-DSA /
// FIPS 206, HQC — selected March 2025) are not in the quantum-vulnerable
// table.

const crypto = require("crypto");

// PQ-safe KEMs/signatures — no findings expected
crypto.generateKeyPair("ml-dsa", {});
crypto.generateKeyPair("ml-kem", {});
crypto.generateKeyPair("fn-dsa", {});
crypto.generateKeyPair("hqc", {});

// crypto.sign with a PQ-safe algorithm name must also stay quiet
crypto.sign("ml-dsa-65", Buffer.from("msg"), privateKey);
crypto.sign("fn-dsa", Buffer.from("msg"), privateKey);

File: tests/fixtures/safe_pq_draft.java
// tests/fixtures/safe_pq_draft.java
//
// Positive non-match fixture for java/pq-vulnerable-crypto.
// KeyPairGenerator/Signature.getInstance with PQ-safe names — including the
// draft NIST standards FN-DSA (FIPS 206) and HQC (5th NIST PQC algorithm,
// selected March 2025) — must NOT fire. classify_java_pq_algo excludes
// FN-DSA/HQC from the WITHDSA fallback explicitly.
import java.security.KeyPairGenerator;
import java.security.Signature;

public class SafePqDraft {
    public void pqSafe() throws Exception {
        // Already PQ-safe — no findings expected
        KeyPairGenerator.getInstance("ML-DSA");
        KeyPairGenerator.getInstance("ML-KEM");
        KeyPairGenerator.getInstance("SLH-DSA");

        // Draft NIST standards — must also be ignored
        KeyPairGenerator.getInstance("FN-DSA");
        KeyPairGenerator.getInstance("HQC");

        Signature.getInstance("SHA256withFN-DSA");
        Signature.getInstance("SHA256withML-DSA");
    }
}
